pylint/checkers/classes/class_checker.py

```diff
@@ -431,7 +442,7 @@ def _called_in_methods(func, klass, methods):
     return False
 
 
-def _is_attribute_property(name, klass):
+def _is_attribute_property(name: str, klass: nodes.ClassDef) -> bool:
```

Comment:
Can't `name` be `None`?

Reply:
There's a check before the call making sure that it's not None:

https://github.com/PyCQA/pylint/blob/6417e8cdee987afb773ad5ab9f5f0e82028c7cac/pylint/checkers/classes/class_checker.py#L1739-L1743

Reply:
`AssignName.name: str | None`. I think it probably shouldn't be `None`, but currently it is typed as such.

Whenever I encounter such issues I tend to leave out the typing as we first need to fix this in `astroid`.